fs/shm/shmfs.c

```diff
@@ -0,0 +1,384 @@
+
+  ret = inode_addref(filep->f_inode);
+  if (ret >= 0)
+    {
+      object = (FAR struct shmfs_object_s *)filep->f_inode->i_private;
+
+      if (object)
+        {
+          ret = shmfs_map_object(object, &entry->vaddr);
+        }
+
+      if (!object || ret < 0)
+        {
+          /* Oops, the file has not been truncated yet or
+           * there was error in up_shmat
+           */
+
+          inode_release(filep->f_inode);
+        }
+      else
+        {
+          entry->munmap = shmfs_munmap;
+          entry->priv.p = (FAR void *)filep->f_inode;
+          mm_map_add(entry);
+        }
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: shmfs_unmap_object
+ ****************************************************************************/
+
+static int shmfs_unmap_area(FAR struct task_group_s *group,
+                            FAR void *vaddr, size_t length)
+{
+  int ret = OK;
+
+#ifdef CONFIG_BUILD_KERNEL
+  unsigned int npages;
+
+  /* Convert the region size to pages */
+
+  if (group)
+    {
+      npages = MM_NPAGES(length);
+
+      /* Unmap the memory from user virtual address space */
+
+      ret = up_shmdt((uintptr_t)vaddr, npages);
+
+      /* Add the virtual memory back to the shared memory pool */
+
+      vm_release_region(get_group_mm(group), vaddr, length);
+    }
+#endif
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: shmfs_munmap
+ ****************************************************************************/
+
+static int shmfs_munmap(FAR struct task_group_s *group,
+                        FAR struct mm_map_entry_s *entry,
+                        FAR void *start,
+                        size_t length)
+{
+  int                      ret;
+
+  /* Partial unmap is not supported yet */
+
+  if (start != entry->vaddr || length != entry->length)
+    {
+      return -EINVAL;
+    }
+
+  /* Unmap the virtual memory area from the user's address space */
+
+  ret = shmfs_unmap_area(group, entry->vaddr, entry->length);
+
+  /* Release the shmfs object. The object gets deleted if no-one has
+   * reference to it (either mmap or open file) and the object has been
+   * unlinked
+   */
+
+  if (ret == OK)
+    {
+      ret = shmfs_release((FAR struct inode *)entry->priv.p);
+    }
+
+  /* Remove the mapping. */
+
+  if (ret == OK)
+    {
+      ret = mm_map_remove(get_group_mm(group), entry);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: shmfs_initialize
+ ****************************************************************************/
+
+void shmfs_initialize(void)
+{
+  register_driver(CONFIG_FS_SHMFS_VFS_PATH, &shmfs_operations, 0666, NULL);
```

Comment:
why need register driver here?

Reply:
You are right, there is no need any longer. This was an artifact from some earlier version. Removed the whole shmfs_initialize.